src/main/java/DavinciBot.java

```diff
@@ -0,0 +1,276 @@
+                echoTask(taskArray);
+            } else {
+                throw new DavinciException("Come on man. Please use: event <description> /from <start> /to <end>");
+            }
+        } else {
+            throw new DavinciException("Whatcha' doing bruh, listen. Please use: event <description> /from <start> /to <end>");
+        }
+        return taskArray;
+    }
+
+    /**
+     * Adds a deadline task to the list, handling the specified description format.
+     *
+     * @param taskArray Array of tasks.
+     * @param description Description of the deadline task.
+     * @return Updated array of tasks.
+     * @throws DavinciException If the deadline format is invalid.
+     */
+    private static Task[] executeDeadlineTask(Task[] taskArray, String description) throws DavinciException {
+        String[] deadlineParts = description.split("/by", SPLIT_INTO_TWO_PARTS);
+        if (deadlineParts.length == 2) {
+            taskArray = Arrays.copyOf(taskArray, taskArray.length + 1);
+            taskArray[taskArray.length - 1] = new Deadline(deadlineParts[0].trim(), deadlineParts[1].trim());
+            echoTask(taskArray);
+        } else {
+            throw new DavinciException("Crappy formatting. Please use: deadline <description> /by <deadline>");
+        }
+        return taskArray;
+    }
+
+    /**
+     * Adds a todo task to the list, handling the specified description format.
+     *
+     * @param taskArray Array of tasks.
+     * @param description Description of the todo task.
+     * @return Updated array of tasks.
+     */
+    private static Task[] executeTodoTask(Task[] taskArray, String description) {
+        taskArray = Arrays.copyOf(taskArray, taskArray.length + 1);
+        taskArray[taskArray.length - 1] = new Todo(description);
+        echoTask(taskArray);
+        return taskArray;
+    }
+
+    /**
+     * Adds a task to the list in a sequential order, and handles the 3 types of commands,
+     * todo, deadline, and event.
+     *
+     * @param userInput User input specifying the task.
+     * @param taskArray Array of tasks.
+     */
+    private static Task[] getTasks(String userInput, Task[] taskArray) {
+        try {
+            Scanner taskScanner = new Scanner(userInput);
+            String taskType = taskScanner.next().toLowerCase();
+
+            if (!taskScanner.hasNext()) {
+                throw new DavinciException("Come on man, specify the " + taskType + " task.");
+            }
+
+            String description = taskScanner.nextLine().trim();
+            switch (taskType) {
+            case "todo":
+                taskArray = executeTodoTask(taskArray, description);
+                break;
+            case "deadline":
+                taskArray = executeDeadlineTask(taskArray, description);
+                break;
+            case "event":
+                taskArray = executeEventTask(taskArray, description);
```

Comment:
consider avoiding the use of magic literals by storing them in static final vars or enums

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers